app.py

```diff
@@ -6,167 +6,205 @@
+            u'email': u'test@example.com',
+            u'given_name': u'gtv2super',
+            u'exp': now + 600,  # Add 10 minutes onto the current time.
+            u'sid': u'b160650a57dd1db6d5af38f7f90d25e3',
+            u'iat': now,
+            u'amr': [u'pwd'],
+            u'nickname': u'test_user',
+            u'nbf': 1511535122,
+            u'sub': u'e463d3c8-42e8-41f7-835b-87c0f722fe4e'
+        }
+
+        id_token = jwt.encode(jwt_payload, private_key_pem, algorithm='RS256')
+
+        token_exchange_payload = {
+            "id_token": id_token.decode('utf-8'),
+            "access_token": str(uuid.uuid4()),
+            "expires_in": 3600,
+            "token_type": "Bearer",
+            "refresh_token": str(uuid.uuid4())
+        }
+
+        return jsonify(token_exchange_payload)
+
+
+    @app.route('/get_key')
+    def get_key():
+        """
+        Return the public JWK in JWKS format.
+        :return: response
+        """
+        public_key_dict = json.loads(public_key)
+        return jsonify({
+            'keys': [
+                public_key_dict
+            ]
+        })
+    
+    return app
+
+class Mirror:
     """
-    Register an endpoint.
+        Handles encapsulating the default Mirror endpoints
     """
-    app.logger.info(request)
-    parsed_json = request.get_json()
-
-    endpoint = parsed_json['endpoint']
-    methods = parsed_json['methods']
-    status_code = parsed_json['status_code']
-    # TODO headers = parsed_json['headers']
-
-    # We will default to a list with an empty response object if no object has been set in list
-    json_responses = [{}] if not parsed_json['json_responses'] else parsed_json['json_responses']
 
-    for rule in app.url_map.iter_rules():
-        if rule.rule == endpoint:
-            # Accessing private isn't nice. Raise PR with Werkzeug to support deleting endpoints?
-            endpoint_responses.pop(endpoint, None)
-            app.url_map._rules.remove(rule)
+    def __init__(self, create_app):
+        """
+        Passed a function that returns a Flask app context
+        """
+        self.create_app = create_app
+        self.app = create_app()
 
-    def func():
-        # Support for pagination
-        if endpoint_responses[endpoint].get('position') > len(endpoint_responses[endpoint].get('responses')) - 1:
-            # If functions calls are over the length of the json responses, return the last response we can
-            positioned_response = endpoint_responses[endpoint].get(
-                'responses')[len(endpoint_responses[endpoint].get('responses')) - 1]
-        else:
-            # Get the json response we want to return on this round
-            positioned_response = endpoint_responses[endpoint].get(
-                'responses')[endpoint_responses[endpoint].get('position')]
+    def get_application(self):
+        """
+        Handler for every call, checks if a reload is needed
+        """
+        global to_reload
```

Comment:
Can you explain the usage of `global` here? I've never used `global` before. Trying to work out if there's a way this can be encapsulated into the class without globals.